Add documentation to README.md (#22)
rachelkberryman/New-Bank-Group-4: README.md

# Application setup and navigation

In this section we will walk you through how to setup and navigate through the New Bank Java application.

## Setup Google Authentication

Before initiating the NewBank app, you will need to install the Google Authentication Application on a mobile device (iOS or Android), when you have completed the installation follow these steps:

1. Tap on `Get Started`
2. Select `Enter a setup key`
3. On the `Account` text field enter any custom name (e.g. MyName)
4. On the `Key` text field enter this exact key: `NY4A5CPJZ46LXZCP`
5. Make sure the account is set on `Time-based` from the drop down menu
6. Tap on the `Add` button

## Executing the NewBank App

In order to run the NewBank Java application, you will first need to make sure you have at least Java SDK 11 installed in your designated IDE. You will then have to run the following commands:

1. Open a terminal from the root of the New-Bank-Group-4 project
2. In the terminal enter the `javac newbank/server/NewBankServer.java` command to compile the code

> 3. Enter the `java newbank/server/NewBankServer.java` command on the same terminal

Reviewer: ```suggestion
3. Enter the `java newbank/server/NewBankServer` command on the same terminal
```
same here when running, no .java ending in command line apparently

4. Then you should get a message on the terminal stating that the New Bank Server is listening to a local port number. Leave this terminal open and move on to the next step
5. Open a second terminal from the root of the New-Bank-Group-4 project
6. Enter the `javac newbank/client/ExampleClient.java ` command

> 7. Enter `java newbank/client/ExampleClient.java ` command on the same second terminal

Reviewer: ```suggestion
7. Enter `java newbank/client/ExampleClient` command on the same second terminal
```
It should be entered without the .java I think. I just tried it and it only worked when not entering .java. (google says so too, but I have no idea ^^)

Reviewer: this only applies to line 70, not 69

3. Enter a new username (e.g. examplecustomer) in the terminal and press enter
4. Enter a new password (e.g. @Test1234)
5. Then you should get a prompt stating `User: 'examplecustomer' Created` and then be redirect to the New Bank Main Menu again
6. Move on to the Logging in section of this documentation

**Do not exit the application or close any terminals if you want to login with the username you just created**

## Logging in

In this intance we are going to demonstrate how to sign in with an existing account with the following step by step guide:

1. From the New Bank Main Menu enter the value `1` in the terminal and press enter
2. Enter an existing username already created (e.g. examplecustomer) and press enter
3. Enter the password that was set for that username (e.g. @Test1234) and press enter
4. If the correct values were submitted, then the application will return a `You are successfully logged in. What do you want to do now?` alongside a set of different commands the customer can choose from.

## Logged in User Protocals & Commands

The following user input and navigation is implemented within the run() method within the server/NewBankClientHandler.java class

> Majority of the methods for these commands are based in the server/NewBankClient.java class, however there are some that are implemented in other classes.

Reviewer: ```suggestion
Majority of the methods for these commands are based in the server/NewBankClient.java and server/NewBank.java classes, however there are some that are implemented in other classes.
```

> 1. Show My Accounts - Java method associated to it: showMyAccounts()

Reviewer: ```suggestion
1. Show My Accounts - Java method associated to it: showMyAccounts() (based within the server/Customer.java class)
```

> 2. Transfer to another user - Java method associated to it: transferToUser()

Reviewer: ```suggestion
2. Transfer to another user - Java method associated to it: transferToUser() (based within the server/NewBank.java class)
```

> 3. Transfer to another owned account - Java method associated to it transferToSelf()

Reviewer: ```suggestion
3. Transfer to another owned account - Java method associated to it transferToSelf() (based within the server/NewBank.java class)
```

> 4. Create New Account - Java method associated to it createLoginAccount()

Reviewer: ```suggestion
4. Create New Account - Java method associated to it createLoginAccount() (based within the server/NewBank.java class)
```

> 5. Close an Account - Java method associated to it  - Java method associated to it: closeAccount() (based within the server/Customer.java class)

Reviewer: ```suggestion
5. Close an Account - Java method associated to it: closeAccount() (based within the server/Customer.java class)
```
duplicate

> 6. Add Funds to an Account - Java method associated to it addMoneyToAccount

Reviewer: ```suggestion
6. Add Funds to an Account - Java method associated to it addMoneyToAccount() (based within the server/NewBank.java class)
```

> 7. Show NewBank Loan Ledger (all customers) - Java method associated to it: addMoneyToAccount()

Reviewer: ```suggestion
7. Show NewBank Loan Ledger (all customers) - Java method associated to it: loansToString() (based within the server/NewBank.java class)
```

> 8. Offer loan - Java method associated to it - Java method associated to it: offerLoan()

Reviewer: ```suggestion
8. Offer loan - Java method associated to it - Java method associated to it: offerLoan() (based within the server/NewBank.java class)
```

> 9. Take out a Loan - Java method associated to it - Java methods associated to it: selectLoan() and run() within the server/NewBankClientHandler.java

Reviewer: ```suggestion
9. Take out a Loan - Java method associated to it - Java methods associated to it: selectLoanToTakeOut() (based within server/NewBankClientHandler.java) & addMoneyToAccount() (based within server/NewBank.java)
```

> 10. Withdraw a Loan Offer - Java method associated to it - selectLoan() and run() within the server/NewBankClientHandler.java

Reviewer: ```suggestion
10. Withdraw a Loan Offer - Java method associated to it - selectMyOfferedLoan() (based within server/NewBankClientHandler.java) & addMoneyToAccount() (based within server/NewBank.java)
```

> 11. Repay a Loan - Java Method associated to it repayLoan() within server/NewBank.java class

Reviewer: ```suggestion
11. Repay a Loan - Java Method associated to it - repayLoan() (based within server/NewBank.java)
```

12. Log out - Conditional within the run() method

### Create New Bank Account command

To create a new bank account follow the below steps:

1. From the list of user commands enter `4` to create a new bank account and press enter
2. Enter `1` to create a current account
3. Enter a customised name you can remember your current account (e.g. example-currect-account) and press enter
4. Enter an amount that you would like to deposit within the account (e.g. 7000) and press enter
5. Then you should get a message similar to this `Successfully created a new Current Account named 'Example Current Account' with an initial balance of 7000.0£`
6. You will then be returned to the list of customer commands.

**Note: When adding a custom name to your bank account, make sure there's no spaces in between the characters**

### Displaying all bank accounts

> For the displaying of bank accounts we have implemented an IBAN method which associatesa customer with a custom bank account id as soon as the customer creates a bank account. To see how the customer can view their bank account/s, please follow the below steps:

Reviewer: ```suggestion
For the displaying of bank accounts we have implemented an IBAN method which associates customer with a custom bank account id as soon as the customer creates a bank account. To see how the customer can view their bank account/s, please follow the below steps:
```
typo

1. From the user command enter `1` 
2. You should now be presented with a list of created bank accounts including their IBAN

> 3. If you wish to transfer or add funds between on account to another (external or to self), please make sure you keep a note of the sender and receiver IBAN

Reviewer: ```suggestion
3. If you wish to transfer funds to this account from an external account (different username), please make sure you keep a note of the sender and receiver IBAN. You will not need the IBAN in order to transfer between accounts linked to the same username.
```
When transferring to self, no IBAN is needed.

4. You will be shortly redirected back to the user logged in menu

### Transfer to another user

1. From the list of user commands enter the value `2`
2. Enter the username of receiver username which has already been created and has a bank account (e.g. receiverUser) and press enter

> 3. Enter the IBAN of the receiver you want to transfer funds to and press enter

Reviewer: ```suggestion
3. Enter the IBAN of the receiver bank account you want to transfer funds to and press enter
```

> 4. Enter your own IBAN number you want to transfer funds from (enter the numbered list value, not the actual IBAN number)

Reviewer: ```suggestion
4. Enter the number next to the account you want to transfer the funds from (enter the numbered list value, not the actual IBAN number)
```

5. Enter the amount you want to transfer 
6. Open you Google Authenticator app and enter the 6 digit number presented to you on the terminal and press enter
7. You should be presented with a similar message like this: `Successfully sent 3.0£ from your account 'ExampleAccount' to account GB24NWBK99999974358045.`
8. You will then be redirected to the user commands menu 

### Loan feature

#### Show NewBank Loan Ledger

> When entering the value `7` from the logged in user commands menu, you will be redirected to a terminal which gives the option to the user to view the offered loans to taken out loans.

Reviewer: ```suggestion
When entering the value `7` from the logged in user commands menu, you will be redirected to a terminal which gives the option to the user to view all offered loans and taken out loans by NewBank customers.
```
a bit more precise
